src/api_test.ts

```diff
@@ -1,142 +1,267 @@
+                  assertEquals(error.status, 429);
+                  assertEquals(error.message, "429: ratelimited");
+                }
+              },
+            );
+
+            MockFetch.reset();
+          },
+        );
 
-          mf.reset();
-        },
-      );
+        await t.step(
+          "should return successful response JSON",
+          async () => {
+            MockFetch.mock("POST@/api/chat.postMessage", () => {
+              return new Response(
+                '{"ok":true, "channel": "C123", "message": {}}',
+              );
+            });
 
-      await t.step(
-        "should throw if response returns an HTTP status code >= 400",
-        async () => {
-          mf.mock("POST@/api/chat.postMessage", () => {
-            return new Response("big explosions", { status: 500 });
-          });
+            const res = await client.apiCall("chat.postMessage", {});
+            assertEquals(res.ok, true);
+            assertEquals(res.channel, "C123");
+            assertEquals(res.message, {});
+
+            MockFetch.reset();
+          },
+        );
 
-          await assertRejects(
+        await t.step("toFetchResponse method", async (t) => {
+          await t.step(
+            "should return usable Response with payload {'ok': false}",
             async () => {
-              return await client.apiCall("chat.postMessage", {});
+              MockFetch.mock("POST@/api/chat.postMessage", () => {
+                return new Response('{"ok":false}', {
+                  headers: { "Retry-After": "120" },
+                });
+              });
+
+              const res = await client.apiCall("chat.postMessage", {});
+              assertEquals(res.ok, false);
+              const fullRes = res.toFetchResponse();
+              assertInstanceOf(fullRes, Response);
+              assertEquals(fullRes.headers?.get("Retry-After"), "120");
+
+              MockFetch.reset();
             },
-            Error,
-            "500: big explosions",
           );
 
-          mf.reset();
-        },
-      );
-
-      await t.step("should return successful response JSON", async () => {
-        mf.mock("POST@/api/chat.postMessage", () => {
-          return new Response('{"ok":true}');
+          await t.step(
+            "should return usable Response with payload {'ok': false}",
+            async () => {
+              MockFetch.mock("POST@/api/chat.postMessage", () => {
+                return new Response('{"ok":false}', {
+                  headers: { "Retry-After": "120" },
+                });
+              });
+
+              try {
+                const res = await client.apiCall("chat.postMessage", {});
+                if (!res.ok) {
+                  const fullRes = res.toFetchResponse();
+                  console.log(fullRes);
+                  console.log(fullRes.headers);
+                }
+              } catch (error) {
```

Comment:
This catch block will never execute, which is why I'm confused as to this test in comparison with the previous test. They are effectively the same test because the `catch` block never executes, and so the assertions are the same. Am I missing something?

Reply:
Ahh I see, yes you are correct I will simplify the test 💯 

This behavior may have be brought over from `copy/paste` the above code

Reply:
Ohh actually this was a duplicate test, not sure how it got there, maybe while doing prototyping